docs/sdks/server-sdks/python.md

```diff
@@ -67,56 +72,62 @@ For the assignment event to send the relevant information, we have to expand the
 We have also stored out SDK Key into the environment variable `EPPO_API_KEY` as a common safety practice.
 
 ```python
-import logging, os
+import logging
+import os
 from uuid import uuid4
 from time import sleep
 import eppo_client
 from eppo_client.config import Config, AssignmentLogger
 
 logging.basicConfig(
-	filename='eppo_assignments.csv',
-	level=logging.INFO,
-	format=f'%(message)s')
+    filename='eppo_assignments.csv',
+    level=logging.INFO,
+    format=f'%(message)s')
+
 
 class LocalAssignmentLogger(AssignmentLogger):
-	def log_assignment(self, assignment):
-		logging.info(assignment)
+    def log_assignment(self, assignment):
+        logging.info(assignment)
+
 
 client_config = Config(api_key=os.getenv("EPPO_API_KEY"),
-	assignment_logger=LocalAssignmentLogger())
+                       assignment_logger=LocalAssignmentLogger())
 eppo_client.init(client_config)
 client = eppo_client.get_instance()
 
 # Give the client some time to initialize.
+# Note that the client may get stuck on this step if there are errors.
+# Please refer to the logs.
```

Comment:
This one really tripped me up at first. This can easily become an infinite loop if there are errors. Those errors ARE written to the logs, but in the console, the user only sees a slew of `"Waiting for the client to initialize"` messages. I added comments to make it clear that if the code gets stuck here, users should check the logs for errors. I'm wondering if there's a better way to handle this though... Maybe we should only try a fixed number of times and then throw an error if the client doesn't initialize after say 10 seconds?

Reply:
I was running this code on an experiment that I had not set up a feature flag for (in my onboarding workspace). The experiment existed, but the experiment configs had `is_enabled` set to False, which caused a method in the source code to return `None` and basically tricked this while loop into thinking that the client was perpetually initializing. I can give you a more detailed explanation of it in a call if you want. It's a bit of a rabbit hole, but it at least illuminated [this unexpected behavior](https://eppo-group.slack.com/archives/C04KR1GGE3C/p1709841852192259).